src/core/namespace.c

```diff
@@ -2446,6 +2460,17 @@ int setup_namespace(
 
         /* MS_MOVE does not work on MS_SHARED so the remount MS_SHARED will be done later */
         r = mount_move_root(root);
+        if (r == -EINVAL && root_directory) {
+                /* If we are using root_directory and we don't have privileges (ie: user manager in a user
+                 * namespace) and the root_directory is already a mount point in the parent namespace,
+                 * MS_MOVE will fail as we don't have permission to change it (of course with EINVAL rather
+                 * than EPERM). Attempt to bind-mount it over itself (like we do above if it's not already
```

Comment:
"of course"? is that mean sarcastically? because if not i can't see why it would be so "of course"? ;-)

Reply:
very much sarcastically :-)

Reply:
maybe add an emoji, to make this clear. 🤯 or so

might not be obvious otherwise.

Reply:
ok, I've just removed it to avoid confusion

Reply:
you missed your chance to add another emoji to the codebase. Sad!

Reply:
Surely you meant: "Sad! 😢"

Reply:
good point 👉!